src/move_ref.rs

```diff
@@ -321,13 +321,50 @@ unsafe impl<'a, T: ?Sized> DerefMove for MoveRef<'a, T> {
   }
 }
 
-unsafe impl<P> DerefMove for Pin<P>
+unsafe impl<'a, T> DerefMove for Pin<&'a T>
 where
-  P: DerefMove,
-  P::Target: Unpin,
+  Pin<&'a T>: DerefMove + DerefMut,
+  Pin<&'a T>: Deref<Target = T>,
+  T: Unpin,
 {
   // SAFETY: We do not need to pin the storage, because `P::Target: Unpin`.
-  type Storage = P::Storage;
+  type Storage = Pin<&'a T>;
+
+  #[inline]
+  fn deref_move<'frame>(
+    self,
+    storage: DroppingSlot<'frame, Self::Storage>,
+  ) -> MoveRef<'frame, Self::Target>
+  where
+    Self: 'frame,
+  {
+    Pin::into_inner(Pin::as_move(Pin::new(self), storage))
+  }
+}
+
+unsafe impl<'a, T> DerefMove for Pin<&'a mut T>
```

Comment:
Do we still even need `Pin::as_move` below?

Reply:
I'm not quite sure what you mean here.

Do you mean:

1. in these two `DerefMove` impls, I should stop using `Pin::as_move` and instead (effectively) inline its contents? (this is a good change as it removes unsafety); or
2. The same, but also entirely delete `PinExt`.

I assume the first, so I've put that in a commit. `PinExt` still appears to be useful because it covers `!Unpin` types. However, it doesn't appear to be required by any tests or doc tests (other than its own) so I'm not sure. Let me know!